internal/subscription/validation.go

```diff
@@ -67,23 +67,21 @@ func GetValidatedModelForCreate(ctx context.Context, req *metrov1.Subscription)
 	}
 
 	// set push auth
-	if req.GetPushConfig() != nil && req.GetPushConfig().GetAttributes() != nil {
-		pushAttr := req.GetPushConfig().GetAttributes()
-
+	if req.GetPushConfig() != nil && req.GetPushConfig().AuthenticationMethod != nil {
+		//fetching basicAuth credentials from push config
+		basicAuth := req.GetPushConfig().GetBasicAuth()
```

Comment:
`GetBasicAuth()` returns nil if the authentication method is not basic auth.
I understand that we only support `basicAuth`. But perhaps it would be better if there is a nil check on `basicAuth` after this.